File: app/src/main/java/com/huhx0015/androidbooster/injections/modules/FragmentModule.java
package com.huhx0015.androidbooster.injections.modules;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import com.huhx0015.androidbooster.injections.scopes.ActivityScope;
import dagger.Module;
import dagger.Provides;

@Module
public class FragmentModule {

    /** CLASS VARIABLES ________________________________________________________________________ **/

    @NonNull
    private Fragment mFragment;

    /** CONSTRUCTOR METHODS ____________________________________________________________________ **/

    public FragmentModule(@NonNull Fragment fragment) {
        this.mFragment = fragment;
    }

    /** MODULE METHODS _________________________________________________________________________ **/

    @Provides
    @NonNull
    @ActivityScope
    Fragment providesFragment() {
        return mFragment;
    }
}

File: app/src/main/java/com/huhx0015/androidbooster/injections/modules/FragmentModule.kt
package com.huhx0015.androidbooster.injections.modules

import androidx.fragment.app.Fragment
import com.huhx0015.androidbooster.injections.scopes.ActivityScope
import dagger.Module
import dagger.Provides

@Module
class FragmentModule (private val fragment: Fragment) {

    /** MODULE METHODS _________________________________________________________________________  */

    @Provides
    @ActivityScope
    fun providesFragment(): Fragment {
        return fragment
    }
}